src/main/kotlin/solve/importer/ProjectParser.kt

```diff
@@ -154,55 +150,47 @@ object ProjectParser {
 
         for (folder in directory.listFiles()) {
             if (folder.name != IMAGE_DIRECTORY_NAME) {
-                parseOutputs(folder, layers, landmarks, errorFolders)
+                partialParseOutputs(folder, outputs, errorFolders, algorithms, layers)
             } else {
                 isImagesExist = true
-                parseImages(folder, images)
+                partialParseImages(folder, images)
             }
         }
 
         if (!isImagesExist) {
+            val ownerWindow = importer.root.scene.window
             createAlertForError("The images folder is missing in the directory", ownerWindow)
             return null
         }
 
-        if (errorFolders.isNotEmpty()) {
-            alertErrorFolderMessage()
-        }
+        images.map { img ->
+            hasAnyErrors = img.errors.isNotEmpty()
 
-        landmarks.forEach { (name, outputs) ->
-            frames.add(ImporterProjectFrame(ProjectFrame(name, Path(images[name]!!), outputs)))
+            frames.add(FrameAfterPartialParsing(img, outputs.filter { img.name == it.name }))
         }
 
-        var hasAnyErrors = false
-        frames.forEach {
-            incorrectExtensionError(it)
-            noSomeAlgorithmError(it, layers)
-            if (it.isImageErrored) {
-                hasAnyErrors = true
-            }
+        hasAnyErrors = noSomeAlgorithmError(frames, layers)
```

Comment:
What if there hasAnyErrors was true, but `noSomeAlgorithmError(frames, layers)` is false?
hasAnyErrors = hasAnyErrors || noSomeAlgorithmError(frames, layers)